### PR TITLE
Pick omit

### PR DESCRIPTION
**Pull request by brijeshnec**

### CODE REVIEW

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java [hunk 1]
```diff
@@ -471,18 +472,66 @@ public static MultiMap getHeadersForRemoteCallFromRegUpdate(List<Map<String, Obj
 
 	public static Uni<RestResponse<Object>> generateEntityResult(List<Object> contextHeader, Context context,
 																 int acceptHeader, Object entity, String geometryProperty, String options, LanguageQueryTerm langQuery,
-																 JsonLDService ldService) {
+																 JsonLDService ldService,List<String> omitList,String pick) {
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java]: Pick parameter should be List of String.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java [hunk 2]
```diff
@@ -471,18 +472,66 @@ public static MultiMap getHeadersForRemoteCallFromRegUpdate(List<Map<String, Obj
...
 			for (Tuple2<String, String> entry : headers) {
 				resp = resp.header(entry.getItem1(), entry.getItem2());
 			}
-			return resp.entity(resultBodyAndHeaders.getItem1()).build();
+			return resp.entity(processPickOmit(resultBodyAndHeaders.getItem1(),pick,omitList)).build();
 		});
 	}
-
+	public static Object processPickOmit(Object object,String pick, List<String> omitList){
+		List<String> pickList = (pick == null) ? new ArrayList<>() : Arrays.asList(pick.split(","));
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java]: Not require to split if you take pick as List of String.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java [hunk 3]
```diff
@@ -890,7 +939,7 @@ public static String getTriggerReason(int triggerReason) {
 
 	public static Uni<RestResponse<Object>> generateQueryResult(HttpServerRequest request, QueryResult queryResult,
 																String options, String geometryProperty, int acceptHeader, boolean count, int limit, LanguageQueryTerm lang,
-																Context context, JsonLDService ldService) {
+																Context context, JsonLDService ldService,List<String> omitList,String pick) {
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java]: Pick parameter should be List of String.

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java [hunk 4]
```diff
@@ -83,32 +80,53 @@ public Uni<RestResponse<Object>> getEntity(HttpServerRequest request, @QueryPara
 			@QueryParam(value = "localOnly") boolean localOnly, @PathParam("entityId") String entityId,
 			@QueryParam(value = "doNotCompact") boolean doNotCompact,
 			@QueryParam("containedBy") @DefaultValue("") String containedBy, @QueryParam("join") String join,
-			@QueryParam("idsOnly") boolean idsOnly, @QueryParam("joinLevel") @DefaultValue("1") int joinLevel) {
+			@QueryParam("idsOnly") boolean idsOnly, @QueryParam("joinLevel") @DefaultValue("1") int joinLevel,
+			@QueryParam("pick") String pick,@QueryParam("omit") String omit) {
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java:84]: Pick and Omit parameter should be List of String.

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java [hunk 5]
```diff
@@ -83,32 +80,53 @@ public Uni<RestResponse<Object>> getEntity(HttpServerRequest request, @QueryPara
...
+		}
 		List<Object> headerContext;
 		headerContext = HttpUtils.getAtContext(request);
 		logger.debug("retrieve called: " + request.path());
 		return HttpUtils.getContext(headerContext, ldService).onItem().transformToUni(context -> {
 //			AttrsQueryTerm attrsQuery;
 			LanguageQueryTerm langQuery;
-			try {
+			String finalPick;
+			List<String> omitList = omit == null ? new ArrayList<>() : Arrays.asList(omit.replaceAll("[\"\\n\\s]", "").split(","));
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java:100]: No need to this logic if you take omit and pick parameter as list of string.

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java [hunk 6]
```diff
@@ -135,7 +153,9 @@ public Uni<RestResponse<Object>> query(HttpServerRequest request, @QueryParam("i
 			@QueryParam("count") boolean count, @QueryParam("containedBy") @DefaultValue("") String containedBy,
 			@QueryParam("join") String join, @QueryParam("idsOnly") boolean idsOnly,
 			@QueryParam("joinLevel") @DefaultValue("1") int joinLevel, @QueryParam("doNotCompact") boolean doNotCompact,
-			@QueryParam("entityMap") String entityMapToken, @QueryParam("maxDistance") String maxDistance, @QueryParam("minDistance") String minDistance, @Context UriInfo uriInfo) {
+			@QueryParam("entityMap") String entityMapToken, @QueryParam("maxDistance") String maxDistance,
+			@QueryParam("minDistance") String minDistance, @Context UriInfo uriInfo,
+			@QueryParam("pick") String pick,@QueryParam("omit") String omit) {
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java:158]: Same as above comment

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/controller/SubscriptionController.java [hunk 7]
```diff
@@ -103,7 +103,7 @@ public Uni<RestResponse<Object>> getSubscriptionById(HttpServerRequest request,
 			return subService.getSubscription(HttpUtils.getTenant(request), subscriptionId).onItem()
 					.transformToUni(subscription -> {
 						return HttpUtils.generateEntityResult(contextHeader, context, acceptHeader, subscription, null,
-								options, null, ldService);
+								options, null, ldService,null,null);
```
Comment (amitrnec) [inline SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/controller/SubscriptionController.java:106]: Add postman test cases for Pick and omit functionality.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java (hunk 1 above)
Comment (brijeshnec) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java]: done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java (hunk 2 above)
Comment (brijeshnec) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java]: done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java (hunk 3 above)
Comment (brijeshnec) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/tools/HttpUtils.java]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java (hunk 4 above)
Comment (brijeshnec) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java:84]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java (hunk 5 above)
Comment (brijeshnec) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java:100]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java (hunk 6 above)
Comment (brijeshnec) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java:158]: done

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/controller/SubscriptionController.java (hunk 7 above)
Comment (brijeshnec) [reply SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/controller/SubscriptionController.java:106]: done

**Pull request merged**